docs/explanation/metrics.md

```diff
@@ -0,0 +1,109 @@
+
+In June 2020 we stopped using Amplitude due to cost.  That leaves us with a lot of things referring to *amplitude events* still.  That's fine for identifying that subset of events, but understand we don't actually use Amplitude anymore.
+
+## So, where does that leave us?
+
+Well, it's a bit complicated and you should make sure you've read our [metrics reference](../reference/metrics) first.
+
+In the `firefox_accounts` dataset in the `mozdata` project:
+
+* `fxa_[server]_*` tables will be most of the amplitude events
+* `fxa_log_[server]_*` tables will be most of the flow events
+
+However, there are a bunch of other tables in there as well.  In short, only experience can really help you find the data you are looking for if there isn't already a dashboard for it.  Ask the FxA team or #data-help in slack.
+
+# How are metrics emitted?
+
+## Auth server
+
+Activity events, flow events and amplitude events are all just mozlog-format log lines. There are functions for each of them in the module `lib/log.js`, respectively called `activityEvent`, `flowEvent` and `amplitudeEvent`. However, to reduce boilerplate in the codebase they’re actually invoked via calls to `request.emitMetricsEvent`. That function is defined in `lib/metrics/events.js` and is decorated onto the request object by some code in `lib/server.js`.
+
+Activity events must have a `uid` property. Some call sites also set `device_id`. Properties containing user agent information and the OAuth client id are set automatically based on the request object.
+
+Flow events have their properties set automatically too, either from the request object or, if there is an authentication token of some kind, from data stored against the uid and token id in memcached. This data is widely referred to as `metricsContext` throughout the codebase and has a lifetime of 2 hours, which is kind of an arbitrary limit that allows us to fetch data for most flows while ignoring dubious data from stale or non-authentic sources. The `metricsContext` is written to memcached via calls to `request.stashMetricsContext` (where it’s copied from the request data) and `request.propagateMetricsContext` (where it’s copied from some other stashed data). Those functions are both defined in `lib/metrics/context.js` and decorated onto the request object in `lib/server.js`.
+
+Amplitude events are not explicitly emitted, because they can all be mapped to from previously-existing activity or flow events. Those mappings are defined by structures called `EVENTS` and `FUZZY_EVENTS` in `lib/metrics/amplitude.js`. Any event matching a string key from `EVENTS` or a regex key from `FUZZY_EVENTS` gets transformed into the corresponding event by common code in the `fxa-shared` package (it’s also used by the content server). Most of the properties on those events are set automatically, although there are still some cases where data must be set explicitly at the call site.
```

Comment:
> Amplitude events are not explicitly emitted, because they can all be mapped to from previously-existing activity or flow events.

This is not _entirely_ true because of Payments.  The front end explicitly target Amplitude by sending events of type `amplitude.${groupName}.${eventName}`.  The back end uses `FUZZY_EVENTS` to regex match those strings.